src/main.zig

```diff
@@ -417,8 +417,10 @@ pub var lastFrameTime = std.atomic.Value(f64).init(0);
 pub var lastDeltaTime = std.atomic.Value(f64).init(0);
 
 var shouldExitToMenu = std.atomic.Value(bool).init(false);
-pub fn exitToMenu(_: usize) void {
+var exitNotification = std.atomic.Value(network.DisconnectType).init(.exit);
+pub fn exitToMenu(exitData: network.DisconnectType) void {
 	shouldExitToMenu.store(true, .monotonic);
+	exitNotification.store(exitData, .monotonic);
```

Comment:
Having two atomics like this is a problem.
The main thread could see the new value of `shouldExitToMenu`, but still see the old value of `exitNotification`, which causes a small chances of seeing an incorrect exit value.

Could you maybe just use one attomic of `?network.DisconnectType` to cover both in one?